ClientRuntimes/NodeJS/lib/filters/exponentialretrypolicyfilter.js

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+var _ = require('underscore');
+var utils = require('../utils');
+
+/**
+ * Determines if the operation should be retried and how long to wait until the next retry.
+ *
+ * @param {number} statusCode The HTTP status code.
+ * @param {object} retryData  The retry data.
+ * @return {bool} True if the operation qualifies for a retry; false otherwise.
+ */
+function shouldRetry(statusCode, retryData) {
+  // Retry on HTTP TimeOut 408
+  if ((statusCode < 500 && statusCode !== 408) || statusCode === 501 || statusCode === 505) {
+    return false;
+  }
+
+  var currentCount = (retryData && retryData.retryCount) ? retryData.retryCount : 0;
+
+  return (currentCount < this.retryCount);
+}
+
+/**
+ * Updates the retry data for the next attempt.
+ *
+ * @param {object} retryData  The retry data.
+ * @param {object} err        The operation's error, if any.
+ * @return {undefined}
```

Comment:
wonder whether this is the convention for java script that using "undefined" if there is no return value. I would just get rid of it